src/linear_algebra/pi_tensor_product.lean

```diff
@@ -378,3 +383,32 @@ end multilinear
 end pi_tensor_product
 
 end semiring
+
+section ring
+namespace pi_tensor_product
+
+open pi_tensor_product
+open_locale tensor_product
+
+variables {ι : Type*} {R : Type*} [comm_ring R]
+variables {s : ι → Type*} [∀ i, add_comm_group (s i)] [∀ i, module R (s i)]
+variables {E : Type*} [add_comm_group E] [semimodule R E]
+
+instance : add_comm_group (⨂[R] i, s i) := semimodule.add_comm_monoid_to_add_comm_group R
```

Comment:
Perhaps worth a doc comment pointing out that unlike `tensor_product`, this requires `comm_ring` instead of `semi_ring`.

Reply:
No, `tensor_product` also has a similar section at the end for the `comm_ring` case (that I more or less copied...)

Reply:
Right, but I changed it on master since you did that :)

Reply:
Oh! I guess I can just do the same thing here then.

Reply:
I don't think you can, at least not if the index set is empty.

Reply:
...or not, because of that extra hidden R tensor factor, which is not an `add_comm_group`. I'll just put in the comment.